### PR TITLE
Add event tracking for sam validate --lint metrics

### PR DESCRIPTION
**Pull request by cdavidxu-hub**
#### Which issue(s) does this change fix?
n/a

#### Why is this change necessary?
We should track the usage of the new '--lint' parameter to compare usage with the base command

#### How does it address the issue?
Adding event tracker to lint method

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [x] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [] Write/update unit tests
- [] Write/update integration tests
- [x] Write/update functional tests if needed
- [x] `make pr` passes
- [x] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by qingchm — APPROVED** [review]
LGTM

File: samcli/commands/validate/validate.py [hunk 1]
```diff
@@ -131,6 +132,7 @@ def _lint(ctx: Context, template: str) -> None:
 
     cfn_lint_logger = logging.getLogger("cfnlint")
     cfn_lint_logger.propagate = False
+    EventTracker.track_event("UsedFeature", "CFNLint")
```
Comment (mndeveci) [inline samcli/commands/validate/validate.py:135]: I think it is miss on our end for not using enums but using strings here. We may update as a follow-up PR

**Review by mndeveci — COMMENTED** [review]
Thanks for the contribution! Any tests we can add to validate this?

**Comment by qingchm** [comment]
> Thanks for the contribution! Any tests we can add to validate this?

I verified for other UsedFeature features there are no tests available either, we can improve by adding integration tests for all of them

**Comment by mndeveci** [comment]
> > Thanks for the contribution! Any tests we can add to validate this?
> 
> I verified for other UsedFeature features there are no tests available either, we can improve by adding integration tests for all of them

I see, but we can still verify this in a unit tests. Something like; when `_lint` method is called, `CFNLint` event must be recorded.

**Review by mndeveci — APPROVED** [review]

**Pull request merged**